pageserver/src/tenant.rs

```diff
@@ -1031,6 +1035,42 @@ impl Tenant {
                 {
                     error!("Failed to clean up uninit marked timeline: {e:?}");
                 }
+            } else if crate::is_delete_mark(&timeline_dir) {
+                // If metadata exists, load as usual, continue deletion
+                // If metadata doesnt exist remove timeline dir and delete mark
+                let timeline_id =
+                    TimelineId::try_from(timeline_dir.file_stem()).with_context(|| {
+                        format!(
+                            "Could not parse timeline id out of the timeline uninit mark name {}",
+                            timeline_dir.display()
+                        )
+                    })?;
+
+                let metadata_path = self.conf.metadata_path(timeline_id, self.tenant_id);
+                if metadata_path.exists() {
+                    // Remote deletion did not finish. Need to resume.
+                    timelines_to_resume_deletion.push((
+                        timeline_id,
+                        load_metadata(self.conf, timeline_id, self.tenant_id)?,
+                    ));
+                    continue;
+                }
+
+                // Missing metadata means that timeline directory should be empty at this point.
+                // Remove delete mark afterwards.
+                // Note that failure during the process wont prevent tenant from successfully loading.
+                if let Err(e) = fs_ext::ignore_absent_files(|| {
+                    fs::remove_dir(self.conf.timeline_path(&timeline_id, &self.tenant_id))
+                })
+                .context("remove deleted timeline dir")
+                .and_then(|_| fs::remove_file(&timeline_dir).context("remove delete mark"))
+                {
+                    warn!(
+                        "cannot clean up deleted timeline dir at: {} error: {:#}",
+                        timeline_dir.display(),
+                        e
+                    );
+                };
```

Comment:
Why is this not another `resume` function in `delete.rs`?

Reply:
Its almost the same as `DeleteTimelineFlow::cleanup_remaining_fs_traces_after_timeline_deletion` with the exception that we know in advance that there is no metadata. I think thats why I didnt unite them. I can do this with a flag `expect_metadata` or something

Reply:
Started doing that, realized that function in DeleteTimelineFlow is async and here we are inside spawn_blocking. Theoretically I can use block_on but AFAIR there were some problems with that (remember @koivunej found something during async walredo related experiments)

Reply:
Kept it as is. Can be adjusted in a follow up.

Reply:
Posting a reply from DM's: 

> it's easy to create loops
> as in the thing you block_on might be spawn_blocking some other stuff, in prod you'll run out of threads, everyone waiting for the next

So staying away from block_on there

Reply:
Mentioned in the follow up list. Though I dont see a clear solution here given the inability to use block_on